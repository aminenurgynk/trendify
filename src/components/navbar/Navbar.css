.navbar {
    background-color: #333;
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: #fff;
    padding: 20px 50px;
  }
  
  .navbar-logo {
    font-size: 30px;
    font-weight: bold;
    color: #333;
  }
  
  .navbar-links a {
    padding: 15px;
    margin: 0 10px;
    text-decoration: none;
    color: #333;
    font-size: 20px;
  }
  
  