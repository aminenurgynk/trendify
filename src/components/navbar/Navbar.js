import { AppBar, IconButton, Menu, MenuItem, Toolbar, Typography } from "@mui/material";
import MenuIcon from '@mui/icons-material/Menu';
import React, { useState } from "react";

const Header = () => {
  const [anchorEl, setAnchorEl] = useState(null);
  const open = Boolean(anchorEl);

  const handleMenu = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };


  const appBarStyle = {
    width: '100%',
    margin: 0, // AppBar'ı ekrana sabitlemek için kenar boşluklarını kaldır
    padding: 0, // İç kenar boşluklarını kaldır
    // AppBar'ın içindeki araç çubuğunu (Toolbar) yanlardan sıkıştırmak için
    '& .MuiToolbar-root': { 
      padding: 0, // Toolbar için iç kenar boşluklarını kaldır
      minHeight: '64px', // Toolbar yüksekliğini ayarla
    },
  };


  return (
    <AppBar position="static" sx={appBarStyle}>
    <Toolbar sx={{ justifyContent: 'space-between', width: '100%' }}>
      <IconButton
        size="large"
        edge="start"
        color="inherit"
        aria-label="menu"
        sx={{ mr: 2 }}
        onClick={handleMenu}
      >
        <MenuIcon />
      </IconButton>
      <Menu
        id="menu-appbar"
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'left',
        }}
        keepMounted
        transformOrigin={{
          vertical: 'top',
          horizontal: 'left',
        }}
        open={open}
        onClose={handleClose}
      >
        <MenuItem onClick={handleClose}>Home</MenuItem>
        <MenuItem onClick={handleClose}>Products</MenuItem>
        <MenuItem onClick={handleClose}>Categories</MenuItem>
      </Menu>
      <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
        Trendify
      </Typography>
    </Toolbar>
  </AppBar>
  );
};

export default Header;
