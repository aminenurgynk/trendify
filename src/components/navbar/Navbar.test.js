// Navbar.test.js

import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from './Navbar';

describe('Navbar', () => {
  it('renders correctly', () => {
    render(
      <Router>
        <Navbar />
      </Router>
    );

    // Check if the brand logo text "Trendify" is in the document
    expect(screen.getByText(/trendify/i)).toBeInTheDocument();

    // Check for navigation links
    expect(screen.getByText(/home/i)).toBeInTheDocument();
    expect(screen.getByText(/products/i)).toBeInTheDocument();
    expect(screen.getByText(/contacts/i)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <Router>
        <Navbar />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
