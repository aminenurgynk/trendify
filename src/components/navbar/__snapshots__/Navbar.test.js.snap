// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div
      class="navbar-logo"
    >
      Trendify
    </div>
    <div
      class="navbar-links"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/"
      >
        Products
      </a>
      <a
        href="/"
      >
        Contacts
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
